contracts/reserve/ExchangeHelper.sol

```diff
@@ -142,23 +142,24 @@ contract ExchangeHelper is DAOUpgradeableContract {
 		} else if (address(_buyWith) == daiAddress) {
 			result = _cdaiMintAndBuy(_tokenAmount, _minReturn, receiver);
 		} else {
-			address[] memory path = new address[](2);
-
-			path[0] = address(_buyWith);
-			path[1] = daiAddress;
 			uint256[] memory swap;
 			if (withETH) {
+				address[] memory path = new address[](2);
+				path[0] = uniswapContract.WETH();
+				path[1] = daiAddress;
 				swap = uniswapContract.swapExactETHForTokens{
 					value: msg.value
 				}(_minDAIAmount, path, address(this), block.timestamp);
```

Comment:
this can be moved into the helper.
in the helper
if(paht[0]===weth) swapexactethfortokens

Reply:
It was like that initially but It wasn't working so I moved back to buy function

Reply:
yeah I moved to helper